encrypted-media-respec.html

```diff
@@ -2101,6 +2101,30 @@
           <p>The following steps are run:</p>
           <ol>
             <li><p>Let the <var>session</var> be the associated <a>MediaKeySession</a> object.</p></li>
+            <li><p>Run the <a def-id="destroy-keys-and-licenses-algorithm"></a> algorithm.</p></li>
+            <li><p>Run the <a def-id="update-key-statuses-algorithm"></a> algorithm on the <var>session</var>, providing an empty sequence.</p></li>
+            <li><p>Run the <a def-id="update-expiration-algorithm"></a> algorithm on the <var>session</var>, providing <code>NaN</code>.</p></li>
+            <li><p>Let <var>promise</var> be the <a def-id="closed"></a> attribute of the <var>session</var>.</p></li>
+            <li><p>Resolve <var>promise</var>.</p></li>
+          </ol>
+        </section>
+        <section id="destroy-keys">
+          <h4>Destroy Keys and Licenses</h4>
+          <p>
+            The Destroy Keys and Licenses algorithm is run when the CDM needs to destroy all the keys and licenses associated with a session that
+            have not been explicitly stored.
+          </p>
+          <p>
+            For sessions of type <a def-id="temporary-session"></a> and <a def-id="persistent-usage-record-session"></a>, this algorithm MUST be
+            executed when the <a>MediaKeySession</a> object becomes inaccessible to the page, if the <a def-id="session-close-algorithm"></a>
```

Comment:
The User Agent needs to run these steps "_when_ the `MediaKeySession` object becomes inaccessible to the page", not _after_. `MediaKeySession` is a platform object visible to Javascript. Becoming inaccessible to the page does not mean the UA has deleted all knowledge of the associated CDM session or other state. Or, you can say that whatever causes it to become inaccessible should also cause this algorithm to be run. The following steps do not involve interaction with the page, so the fact the object is no longer JS-visible is not an issue.